# Урок 1
Случайные события. Условная вероятность. Формула Байеса. Независимые испытания
Домашнее задание

In [1]:
import numpy as np

Задача 1

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

In [2]:
# Сначала посчитаем количество способов (сочетаний), которыми можно вытащить 4 карты: 
def combinations(n: int, k: int) -> int:
    """Число сочетаний"""
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k))
fours = combinations(52, 4)
print(fours)

270725.0


In [3]:
# "Хотя бы один туз" означает, что в вынутых картах может быть 1, 2, 3 или 4 туза.
# Посчитаем каждый случай отдельно:
t1 = combinations(4, 1)*combinations(48, 3)
t2 = combinations(4, 2)*combinations(48, 2)
t3 = combinations(4, 3)*combinations(48, 1)
t4 = 1
print(t1+t2+t3+t4)

76145.0


In [4]:
# Удобнее для решения задачи посчитать число противоположных исходов и отнять их от общего количества.
# Из 48 карт, не являющихся тузами, количество способов вынуть 4 карты:
not_t = combinations(48, 4)
print(not_t)

194580.0


In [5]:
# Соответственно, число исходов, соответствующих тому, что был вытянут хотя бы один туз:
t = fours - not_t
print(t)

76145.0


Ответ: 76145 исходов, соответствующих тому, что был вытянут хотя бы один туз.

Задача 2

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?

In [6]:
def permutations(n: int) -> int:
    """Число перестановок"""
    return np.math.factorial(n)
# Посчитаем число способов сесть для семи человек:
n = permutations(7)
print(n)

5040


In [7]:
# Для семи человек всего пар соседних мест - 6. 
# Каждая пара человек может сесть на эти места двумя способами, а остальные 5 любыми способами (перестановки, 5!).
# Соответственно, число способов сесть рядом для двух определённых людей:
m2 = 6*2*permutations(5)
# Вероятность того, что два определенных человека будут сидеть рядом:
print(m2/n)  # это 2/7

0.2857142857142857


In [8]:
# Так как последовательность рассадки внутри тройки в условии не определена, то по аналогии:
# троек мест рядом на скамейке 5,
# каждые три человека могут сесть на места шестью (3!) способами, а все остальные 24 способами (4!).
m3 = 5*6*24
# Вероятность того, что три определенных человека будут сидеть рядом:
print(m3/n) # это 1/7

0.14285714285714285


Ответ: вероятность того, что рядом будут сидеть 
        а) два определённых человека: 2/7 или 0.2857142857142857
        б) три определённых человека: 1/7 или 0.14285714285714285

Задача 3

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

In [9]:
# Для первого вопроса вероятность того, что студент его знает, 5/6.
# Для второго, при условии, что студент знает первый: 49/59
# Для третьго, при условии, что студент знает первый и второй: 48/58

In [10]:
#  Соответственно, вероятность того, что студент знает одновременно три вопроса:
p3 = (5*49*48)/(6*59*58)
print(p3)

0.5727644652250146


In [11]:
#  Соответственно, вероятность того, что студент знает одновременно два вопроса:
p2 = (5*49)/(6*59)
print(p2)

0.692090395480226


Ответ: вероятность того, что среди трёх наугад выбранных вопросов студент знает
    а) все: 0.5727644652250146
    б) два: 0.692090395480226

Задача 4

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

In [12]:
# Считаем, что 𝑃(𝐴)=𝑃(𝐵)=1/2. Посчитаем совместную вероятность P(AB). Событие 𝐴𝐵 соответствует появлению чисел 4 и 6,
# т.е. P(AB)=1/3
# Следовательно, 
# 𝑃(𝐴|𝐵) = P(AB)/𝑃(𝐵) = (1/3)/(1/2) = 2/3
# Поскольку 𝑃(𝐴|𝐵) не равно 𝑃(𝐴), заключаем, что события A и B зависимы.

Ответ: события А и В  не являются независимыми.

Задача 5

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

In [13]:
# При положительном тесте возможны три варианта:
# 1) вы входите в группу тех, кого поражает болезнь, и вы действительно больны. Вероятность этого 0,001*0,99 = 0,00099
# 2) вы входите в группу тех, кого поражает болезнь, но вы не больны, тест ложный. Вероятность этого 0,001*0,01 = 0,00001
# 3) вы не входите в группу тех, кого поражает болезнь, и вы не больны, тест ложный. Вероятность этого 0,999*0,01 = 0,00999

In [14]:
# Вычислим вероятность варианта 1, т.е, что вы действительно больны при положительном тесте:
p = 0.00099/(0.00099+0.00001+0.00999)
print(p)

0.09008189262966333


Ответ: вероятность того, что вы действительно больны: 0.09008189262966333